### DISCUSSION TITLE
Migrating to etcd seems to have broken my cluster

### DISCUSSION THREAD

**Discussion by 2fst4u** [k3s-io/k3s#4585] [Q&A] [answered]
This is 99% my error but I hope there's a way out of it.

I attempted to migrate my cluster from SQLlite to etcd in accordance with https://github.com/k3s-io/k3s/issues/3982 and initially this appeared to have worked. After a while I could run `kubectl get nodes` and see this first master up and running with etc.

Then I tried adding a second master and this is where everything went wrong. The second master complained about bad TLS handshakes which I couldn't progress past. 

After rebooting nodes and restarting services it now appears my first master is broken. The K3S service is in a constant loop and I can't access kubectl commands as it returns `The connection to the server 127.0.0.1:6443 was refused - did you specify the right host or port?`. 

Trying to run the whole server command again, even after stopping the K3S service, results in this alarming error `panic: bootstrap data already found and encrypted with different token` but the token I'm using is straight from `/var/lib/rancher/k3s/server/token`.

I ddin't make this an issue because this is likely my fault and not something that needs fixing. What can I do from here?

**Comment by brandond** [comment, marked as answer]
Can you provide the k3s service logs, going back to the original startup after adding the --cluster-init flag?

At the very least you should be able to roll back to the old sqlite datastore; the file is still there - you'd just need to delete the etcd db directory, remove the cli flag, and rename the sqlite database back into place.

**Reply by 2fst4u** [reply]
Sorry I forgot I made this thread in the heat of panic. Unfortunately I decided to just start fresh and buy some SSDs for booting my Pis so I can run etcd better and without concern of burning out my SD cards. I know that's unsatisfying because it means we can't get to the bottom of it but hopefully I'm an outlier and this migration doesn't break for others.
